### DISCUSSION TITLE
How to track a schedule task?

### DISCUSSION THREAD

**Discussion by 1393467261** [apache/skywalking#8134] [Q&A] [answered]
I am trying to build a span for my schedule task. The task is init by service itself, so there is no active span while processing. Is `@Trace` and `ActiveSpan` still work? If not, is it possible to track a schedule task?

**Comment by wu-sheng** [comment, marked as answer]
I think the supported list has covered many scheduler frameworks. Take a look first.

**Reply by 1393467261** [reply]
I see, skywalking covered lots of scheduler libs, but my scheduler is self-developed

**Reply by wu-sheng** [reply]
Then you need a new plugin, which is normal case. We have provided dev document for that case.

**Reply by 1393467261** [reply]
Fine, the core principle of sky is plugin-oriented instread of exposing too much, right?

**Reply by wu-sheng** [reply]
I am not sure what is `exposing too much`. But yes, we prefer plugin mechanism, from agent to backend.

**Comment by 1393467261** [comment]
I mean allowing user to build span themselves for flexibility. Is it a property way in skywalking?

**Reply by wu-sheng** [reply]
It is always there. What do you exact need? Confused.

**Reply by 1393467261** [reply]
Oh, I'm blind😒, forget about it.
